rotkehlchen/tests/unit/decoders/test_cowswap.py

```diff
@@ -879,6 +879,130 @@ def test_gnosis_vested_claim(database, gnosis_inquirer, gnosis_accounts):
+            notes=f'Burned {gas_amount} XDAI for gas',
+            tx_hash=tx_hash,
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.GNOSIS,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.SPEND,
+            asset=A_XDAI,
+            balance=Balance(amount=FVal(payment_amount)),
+            location_label=user_address,
+            notes=f'Pay {payment_amount} XDAI to claim vCOW',
+            tx_hash=tx_hash,
+            counterparty=None,
+            address=A_GNOSIS_VCOW.resolve_to_evm_token().evm_address,
+        ), EvmEvent(
+            sequence_index=2,
+            timestamp=timestamp,
+            location=Location.GNOSIS,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.RECEIVE,
+            asset=A_GNOSIS_VCOW,
+            balance=Balance(amount=FVal(claim_amount)),
+            location_label=user_address,
+            notes=f'Claim {claim_amount} vCOW from cowswap airdrop',
+            tx_hash=tx_hash,
+            counterparty=CPT_COWSWAP,
+            address=ZERO_ADDRESS,
+            extra_data={AIRDROP_IDENTIFIER_KEY: 'cow_gnosis'},
+        ),
+    ]
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('gnosis_accounts', [['0x0b297C31d2DA6d959Bc911413990653e19F0e283']])
+def test_gnosis_claim_airdrop_with_gno_payment(database, gnosis_inquirer, gnosis_accounts):
+    user_address = gnosis_accounts[0]
+    tx_hash = deserialize_evm_tx_hash('0xdae21fd2a64756326ba0bf119b8ee33cf41480fb758d0d7f17168fcc01622da1')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=gnosis_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp, gas_amount, claim1_amount, payment_amount, claim2_amount = TimestampMS(1645776935000), '0.000259548002076384', '3559.387459031416241036', '2.085578589276220452', '3559.387459031416239786'  # noqa: E501
+    assert events == [
+        EvmEvent(
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.GNOSIS,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_XDAI,
+            balance=Balance(amount=FVal(gas_amount)),
+            location_label=user_address,
+            notes=f'Burned {gas_amount} XDAI for gas',
+            tx_hash=tx_hash,
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            sequence_index=10,
+            timestamp=timestamp,
+            location=Location.GNOSIS,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.AIRDROP,
+            asset=A_GNOSIS_VCOW,
+            balance=Balance(amount=FVal(claim1_amount)),
+            location_label=user_address,
+            notes=f'Claim {claim1_amount} vCOW from cowswap airdrop',
```

Comment:
So in this transaction we have 1 normal airdrop claim and 1 we pay for?

Reply:
That seems to be the case as far as I can tell, but I'm certainly not an expert on understanding exactly what is going on in a transaction like this. You probably know better than I do.

Reply:
I have seen many transactions but for each protocol you study, by studying the code of the contract and going in the discord/forums of the protocol it is your job to also understand what happens, and write the decoder correctly.

I also would not know how each protocol works.